pkg/platform/runtime2/setup/implementations/camel/metadata.go

```diff
@@ -0,0 +1,150 @@
+package camel
```

Comment:
Changes compared to `pkg/platform/runtime/metadata.go`:

```
--- pkg/platform/runtime2/setup/implementations/camel/metadata.go	2021-03-18 14:30:14.202666220 -0700
+++ pkg/platform/runtime/metadata.go	2021-03-18 10:08:20.022254760 -0700
@@ -1,10 +1,9 @@
-package camel
+package runtime
 
 import (
 	"encoding/json"
 	"os"
 	"path/filepath"
-	"strings"
 
 	"github.com/ActiveState/cli/internal/constants"
 	"github.com/ActiveState/cli/internal/errs"
@@ -26,8 +25,8 @@
 
 // MetaData is used to parse the metadata.json file
 type MetaData struct {
-	// InstallDir is the root directory of the artifact files that we need to copy on the user's machine
-	InstallDir string
+	// Path is the directory containing the meta file
+	Path string
 
 	// AffectedEnv is an environment variable that we should ensure is not set, as it might conflict with the artifact
 	AffectedEnv string `json:"affected_env"`
@@ -60,9 +59,9 @@
 }
 
 // InitMetaData will create an instance of MetaData based on the metadata.json file found under the given artifact install dir
-func InitMetaData(rootDir string) (*MetaData, error) {
+func InitMetaData(installDir string) (*MetaData, error) {
 	var metaData *MetaData
-	metaFile := filepath.Join(rootDir, "support", constants.RuntimeMetaFile)
+	metaFile := filepath.Join(installDir, constants.RuntimeMetaFile)
 	if fileutils.FileExists(metaFile) {
 		contents, err := fileutils.ReadFile(metaFile)
 		if err != nil {
@@ -81,20 +80,8 @@
 		metaData.Env = map[string]string{}
 	}
 
-	var relInstallDir string
-	installDirs := strings.Split(constants.RuntimeInstallDirs, ",")
-	for _, dir := range installDirs {
-		if fileutils.DirExists(filepath.Join(rootDir, dir)) {
-			relInstallDir = dir
-		}
-	}
-
-	if relInstallDir == "" {
-		logging.Debug("Did not find an installation directory relative to metadata file.")
-	}
-
-	metaData.InstallDir = relInstallDir
-	err := metaData.Prepare(filepath.Join(rootDir, relInstallDir))
+	metaData.Path = installDir
+	err := metaData.Prepare()
 	if err != nil {
 		return nil, err
 	}
@@ -120,11 +107,11 @@
 	return metaData, nil
 }
 
-func (m *MetaData) hasBinaryFile(root string, executable string) bool {
+func (m *MetaData) hasBinaryFile(executable string) bool {
 	for _, dir := range m.BinaryLocations {
 		parent := ""
 		if dir.Relative {
-			parent = root
+			parent = m.Path
 		}
 		bin := filepath.Join(parent, dir.Path, executable)
 		if fileutils.FileExists(bin) {
@@ -142,10 +129,6 @@
 		logging.Debug("Not setting PYTHONPATH as the user already has it set")
 	}
 
-	// This is broken for two reasons:
-	// 1. Checking in the OS environment will only happen on installation, but at a later point, the OS environment might have changed, and we will overwrite the user's choice here
-	// 2. python code does not need to depend on PYTHONIOENCODING as pointed out here: https://stackoverflow.com/a/9942822
-	// Follow up story is here: https://www.pivotaltracker.com/story/show/177407383
 	if os.Getenv("PYTHONIOENCODING") == "" {
 		m.Env["PYTHONIOENCODING"] = "utf-8"
 	} else {
```